src/qopenglwebpage.cpp

```diff
@@ -186,6 +188,17 @@ void QOpenGLWebPage::drawUnderlay()
 */
 void QOpenGLWebPage::drawOverlay(const QRect &rect)
 {
+    QList<QWeakPointer<QMozGrabResult> >::const_iterator it = mGrabResultList.begin();
+    for (; it != mGrabResultList.end(); ++it) {
+        QSharedPointer<QMozGrabResult> result = it->toStrongRef();
+        if (result) {
+            result->captureImage(rect);
+        } else {
+            qWarning() << "QMozGrabResult freed before being realized!";
+        }
+    }
+    mGrabResultList.clear();
+
     Q_EMIT afterRendering(rect);
```

Comment:
Well, not really. You can't use direct connection in case the signal target lives on another thread and can be removed at any point in time before the slot is actually executed. In our case we don't have any gurantee that the grab result instance won't be removed on UI thread before compositor thread calls QOpenGLWebPage::drawOverlay.

Side note, we also need to add a mutex synchronizing access to mGrabResultList. We append elements to the list from UI thread and clear it from compositor thread. I'll push another patch later.